### DISCUSSION TITLE
Does organic maps offer something in terms of optimizing the routes with multiple points stops?

### DISCUSSION THREAD

**Discussion by iamshreeram** [organicmaps/organicmaps#7970] [Q&A] [answered]
Is it possible to find the most efficient and optimized route to reach all destinations in the given list of co-ordinates, ensuring that I can get to each destination, only once in the shortest possible time and with maximum efficiency?

**Comment by patepelo** [comment]
Do you mean for each stop to be ordered automatically? In that case no, and it's not planned as far as I know.

**Reply by iamshreeram** [reply]
Is it possible to automatically calculate the most efficient route for visiting all destinations in a given list of coordinates or places? For instance, using Organic maps or a similar tool to determine the optimal path where each destination is visited only once and with the shortest distance traveled. 

This feature will help in various use cases like travel planning, delivery services, emergency responders, field services, tour/event coordination, sales representatives, and many more business.

**Comment by auge8472** [comment]
Good luck with the „*most efficient* route“: [traveling salesman problem](https://en.wikipedia.org/wiki/Travelling_salesman_problem). 😁

**Reply by iamshreeram** [reply]
@auge8472 , Yeah, TSP is the problem statement. Is there a plugin or wrapper or any existing solution that addresses this same issue? With OM being around for a while, I assumed someone would have already developed a solution.

If not, would it be beneficial to download the map of a particular location for local use in order to create routes between multiple stops? We could start by creating a wrapper that uses brute force to find the optimized route, and then explore more efficient methods later on.

**Comment by hb0nd** [comment, marked as answer]
see https://github.com/organicmaps/organicmaps/issues/7213

**Reply by iamshreeram** [reply]
Hi @RedAuburn , Thank you for sharing this. Yes, this is my ask. Unlike what Bernie mentioned, I appreciate how OM optimizes the route with fewer stops. However, it would be even better if this feature could be extended to accommodate a greater number of stops.

As Tobia Alberti suggested, the preference for route optimization could be customizable in each user's settings.

**Note** : It would be a great feature if we could consistently generate the optimal path when multiple stops are added, ensuring that the results are consistent across multiple iterations.

**Reply by iamshreeram** [reply]
@RedAuburn , I'd like to use the SDK to find the most efficient route from OM based on a list of coordinates? I am interested in using Python or Rust to achieve this. 

If it's not possible or SDK is not available, what alternative method would you recommend for programmatically obtaining the optimized route from OM?
